src/library/styles/createStyledComponent.js

```diff
@@ -10,10 +10,10 @@ type Element =
   | React$ComponentType<*>
   | string;
 
-type Styles =
-  | Object
-  | Array<Object>
-  | ((props: Object, context?: Object) => Object | Array<Object>);
+type StylesIn = Object | StylesFn | Array<StylesIn>;
```

Comment:
Rather than introducing support for arrays of style functions, could this use case not be solved without modifications by using a single function that returns an array of style objects?  If not, and if this is no longer needed by your new components, I would prefer to hold off on these changes until they are truly needed.